# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineFrame.h
# Commit: 58ea17fc315e
# Full Hash: 58ea17fc315ef8bc3a325a958953a50e047f4499
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-07-30 21:50:30
# Regressor Bug: 1567438
# File Overlap Count: 2
# Description:
#   Bug 1567438 part 1 - Replace the enter-monitor-chain bailout code with type monitoring in C++. r=tcampbell,iain
#   
#   When we resumeAfter a JOF_TYPESET op, we had code to enter the type monitor
#   chain for the bailing op, passing it the top stack value. It's simpler to do
#   this monitoring in C++ in FinishBailoutToBaseline and then treat it as a regular
# ==============================================================================

diff -r cd58aae7d47b -r 58ea17fc315e js/src/jit/BaselineFrame.h
--- a/js/src/jit/BaselineFrame.h	Tue Jul 30 15:12:39 2019 +0000
+++ b/js/src/jit/BaselineFrame.h	Tue Jul 30 12:08:48 2019 +0000
@@ -146,6 +146,12 @@
     return (Value*)this - (slot + 1);
   }
 
+  Value topStackValue() const {
+    size_t numSlots = numValueSlots();
+    MOZ_ASSERT(numSlots > 0);
+    return *valueSlot(numSlots - 1);
+  }
+
   Value& unaliasedFormal(
       unsigned i, MaybeCheckAliasing checkAliasing = CHECK_ALIASING) const {
     MOZ_ASSERT(i < numFormalArgs());